edge/pkg/common/dbm/db_test.go

```diff
@@ -1,97 +1,116 @@
+func TestRegisterModel(t *testing.T) {
+	tests := []struct {
+		name       string
+		modulename string
+		m          interface{}
+	}{
+		{
+			//Failure Case
+			name:       "TestRegisterModel-UnregisteredModule",
+			modulename: "testmodule",
+			m:          "",
+		},
+		{
+			//Success Case
+			name:       "TestRegisterModel-RegisteredModule",
+			modulename: "twin",
+			m:          new(TestDevice),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			RegisterModel(test.modulename, test.m)
+		})
 	}
-	log.LOGGER.Debugf("Delete affected Num: %d", num)
-	return nil
-}
-
-// UpdateMeta update meta
-func UpdateMeta(meta *TestMeta) error {
-	num, err := dbm.DBAccess.Update(meta) // will update all field
-	log.LOGGER.Debugf("Update affected Num: %d, %s", num, err)
-	return err
 }
 
-// UpdateMetaField update special field
-func UpdateMetaField(key string, col string, value interface{}) error {
-	num, err := dbm.DBAccess.QueryTable(MetaTableName).Filter("key", key).Update(orm.Params{col: value})
-	log.LOGGER.Debugf("Update affected Num: %d, %s", num, err)
-	return err
+// TestIsNonUniqueNameError is function to test IsNonUniqueNameError().
+func TestIsNonUniqueNameError(t *testing.T) {
+	tests := []struct {
+		name     string
+		err      error
+		wantBool bool
+	}{
+		{
+			name:     "Suffix-are not unique",
+			err:      errors.New("The fields are not unique"),
+			wantBool: true,
+		},
+		{
+			name:     "Contains-UNIQUE constraint failed",
+			err:      errors.New("Failed-UNIQUE constraint failed"),
+			wantBool: true,
+		},
+		{
+			name:     "Contains-constraint failed",
+			err:      errors.New("The input constraint failed"),
+			wantBool: true,
+		},
+		{
+			name:     "OtherError",
+			err:      errors.New("Failed"),
+			wantBool: false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			gotBool := IsNonUniqueNameError(test.err)
+			if gotBool != test.wantBool {
+				t.Errorf("IsNonUniqueError() failed, Got = %v, Want = %v", gotBool, test.wantBool)
+			}
+		})
+	}
 }
 
-// UpdateMetaFields update special fields
-func UpdateMetaFields(key string, cols map[string]interface{}) error {
-	num, err := dbm.DBAccess.QueryTable(MetaTableName).Filter("key", key).Update(cols)
-	log.LOGGER.Debugf("Update affected Num: %d, %s", num, err)
-	return err
+// TestCleanUp() is functioj to test CleanUp().
+func TestCleanup(t *testing.T) {
+	t.Run("CleanUpTest", func(t *testing.T) {
+		Cleanup()
+		_, err := os.Stat(defaultDataSource)
+		if os.IsExist(err) {
+			t.Error("CleanUp failed ,File not removed")
+		}
+	})
 }
 
-// QueryMetaByKey query meta
-func QueryMetaByKey() error {
-	return nil
+// TestCleanDBFile is function to test cleanDBFile().
+func TestCleanDBFile(t *testing.T) {
+	tests := []struct {
+		name     string
+		filename string
+	}{
+		{
+			// Checks for the negative scenario of CleanBDFile where an unknown file is passed. Positive scenario is handled in CleanUp().
```

Comment:
What is the difference between the test case of TestCleamDBFile and TestCleanup ?? The error returned while deleting an unknown file should be different ?

Reply:
The test case of CleanUp by default takes the dataSource as the parameter. The test case of CleanDBFile takes an external filename and tests the negative scenario where the passed file is not present.